common/sky/moon.py

```diff
@@ -0,0 +1,481 @@
+import numpy as np
+from astropy.time import Time
+from astropy.coordinates import SkyCoord, EarthLocation, Distance, GeocentricTrueEcliptic, Angle
+import astropy.units as u
+from astropy.time import TimeDelta
+from common.sky.utils import local_sidereal_time, current_geocent_frame, geocentric_coors, hour_angle_to_angle
+from common.sky.constants import J2000, EQUAT_RAD
+from common.sky.altitude import Altitude
+from typing import NoReturn, Tuple
+
+
+class Moon:
+
+    def __init__(self):
+
+        #TODO: skip constructor altogether and just use the at method?
+        self.time = None
+        self.time_jd = None
+        self.time_ttjd = None  # important to use correct time argument for this!!
+        self.scalar_input = False
+        self.PIE = None
+        self.LAMBDA = None
+        self.BETA = None
+
+    def at(self, time: Time):
+
+        self.time = time
+        self.time_jd = np.asarray(time.jd)
+        self.time_ttjd = np.asarray(time.tt.jd)  # important to use correct time argument for this!!
+        self.scalar_input = False
+        if self.time_ttjd.ndim == 0:
+            self.time_ttjd = self.time_ttjd[None]  # Makes 1D
+            self.scalar_input = True
+        
+        return self
+
+    def _low_precision_calculations(self) -> NoReturn:
+        """
+        Compute low precision values for the moon location method calculations.
+        """
+        #jd = np.asarray(self.time.jd)
+        T = (self.time_jd - J2000) / 36525.  # jul cent. since J2000.0
+
+        lambd = (218.32 + 481267.883 * T
+                 + 6.29 * np.sin(np.deg2rad(134.9 + 477198.85 * T))
+                 - 1.27 * np.sin(np.deg2rad(259.2 - 413335.38 * T))
+                 + 0.66 * np.sin(np.deg2rad(235.7 + 890534.23 * T))
+                 + 0.21 * np.sin(np.deg2rad(269.9 + 954397.70 * T))
+                 - 0.19 * np.sin(np.deg2rad(357.5 + 35999.05 * T))
+                 - 0.11 * np.sin(np.deg2rad(186.6 + 966404.05 * T)))
+        self.LAMBDA = np.deg2rad(lambd)
+
+        beta = (5.13 * np.sin(np.deg2rad(93.3 + 483202.03 * T))
+                + 0.28 * np.sin(np.deg2rad(228.2 + 960400.87 * T))
+                - 0.28 * np.sin(np.deg2rad(318.3 + 6003.18 * T))
+                - 0.17 * np.sin(np.deg2rad(217.6 - 407332.20 * T)))
+        self.BETA = np.deg2rad(beta)
+
+        pie = (0.9508 + 0.0518 * np.cos(np.deg2rad(134.9 + 477198.85 * T))
+                + 0.0095 * np.cos(np.deg2rad(259.2 - 413335.38 * T))
+                + 0.0078 * np.cos(np.deg2rad(235.7 + 890534.23 * T))
+                + 0.0028 * np.cos(np.deg2rad(269.9 + 954397.70 * T)))
+        self.PIE = np.deg2rad(pie)
+
+    def _high_precision_calculations(self) -> NoReturn:
```

Comment:
Again, I wonder if this is something we always have to do or we can get away only doing a few times .